### PR TITLE
Run lint and test for Tanzu CLI plugins

### PR DESCRIPTION
**Pull request by codegold79**
# Description
_Include a summary of the change. If this change fixes an issue, indicate that on the `Fixes` line. Briefly include
relevant context and motivation._

Enable running linter and test (vet and fmt) for Tanzu CLI plugins.

Fixes #{issue}

## Change Details
Check all that apply:
- [x] New feature 
- [ ] Bug fix 
- [ ] Non-code change 
- [ ] Non-breaking change 
- [ ] Breaking change
- [ ] Requires a documentation change

## Release Note

```release-note
Enable running linter and test (vet and fmt) for Tanzu CLI plugins.
```

# How Has This Been Tested?
_Describe any testing done to verify changes. Provide enough detail that tests can be reproduced._

# Checklist:
- [ ] My code follows the [Contribution and Style guidelines](../../CONTRIBUTING.md) of this project
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added test details that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by yharish991 — CHANGES_REQUESTED** [review]

File: examples/multi-module-integration/cmd/plugin/plugin-demo-bar/main.go [hunk 1]
```diff
@@ -1,3 +1,6 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
```
Comment (rajathagasthya) [inline examples/multi-module-integration/cmd/plugin/plugin-demo-bar/main.go:1]: `s/2021/2023`

File: examples/multi-module-integration/cmd/plugin/plugin-demo-foo/main.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
```
Comment (rajathagasthya) [inline examples/multi-module-integration/cmd/plugin/plugin-demo-foo/main.go:1]: `s/2022/2023`

File: examples/multi-module-integration/cmd/plugin/plugin-demo-bar/main.go (hunk 1 above)
Comment (codegold79) [reply examples/multi-module-integration/cmd/plugin/plugin-demo-bar/main.go]: done

File: examples/multi-module-integration/cmd/plugin/plugin-demo-foo/main.go (hunk 2 above)
Comment (codegold79) [reply examples/multi-module-integration/cmd/plugin/plugin-demo-foo/main.go]: done

File: templates/Makefile [hunk 3]
```diff
@@ -192,20 +192,22 @@ plugin-builder-install:
...
-	--target cli-plugin-build \
-	--output build \
-	--build-arg IMGPKG_VERSION=$(IMGPKG_VERSION) \
-	--build-arg CLI_PLUGIN_VERSION=$(CLI_PLUGIN_VERSION) \
-	--build-arg OCI_REGISTRY=$(REGISTRY_ENDPOINT) \
-	--build-arg CLI_PLUGIN=$* \
-	--build-arg COMPONENT=cmd/plugin/$*
+	$(MAKE) COMPONENT=cmd/plugin/$* lint
+	$(MAKE) COMPONENT=cmd/plugin/$* test
+	# $(DOCKER) build --network=host . -f Dockerfile \
```
[the remark this answers is not in the document]
Comment (codegold79) [reply templates/Makefile]: good catch. done.

**Review by yharish991 — APPROVED** [review]

**Review by rajathagasthya — APPROVED** [review]

**Pull request merged**